apps/service_providers/views.py

```diff
@@ -48,8 +55,11 @@ def matches_blocking_deletion_condition(obj):
 
 
 @require_http_methods(["DELETE"])
+@login_and_team_required
 def delete_service_provider(request, team_slug: str, provider_type: str, pk: int):
     provider = ServiceProvider[provider_type]
+    if not request.user.has_perm(provider.get_permission("delete")):
```

Comment:
Why not use the `@permission_required` decorator here?

Reply:
Because the permission is dependent on the service provider type.